src/Products.js

```diff
@@ -0,0 +1,29 @@
+import useFetch from "./hooks/useFetch";
+import Loading from "./components/Loading"
+import { Link } from 'react-router-dom';
+
+const Products = () => {
+    const {data, loading, error} = useFetch('products');
+    
+    if(loading){
+        //return <Loading />;
+        return <Loading />
+    }
+
+    if(error){
+        return <p>Error Loading Products</p>
+    }
+
+    return (
+        <div className="product-list">
+          {data.length && 
+                data.map((element) => {
+               return <div key={element.id} className="box-container">
```

Comment:
This could be its own component.
```
<div key={element.id} className="box-container">
   <img className="preview-image" src={element.image} alt={element.id} />
   <Link to={`/${element.id}`}>{element.title}</Link>
</div>;
```

Reply:
Tried. Failed. Passing the fetch data as a prop to this as a component broke the program, but didn't give me errors, so I left as is for Midterm purposes.

Reply:
Replace the snippet I shared with this. Disclaimer: Could be small errors here since I did not write this out in an IDE.

```
<ProductListItem id={element.id} src={element.image} title={element.title} />
```

Your new custom component would look like this:
```
const ProductListItem = (props) => {
   return (
      <div key={props.id} className="box-container">
         <img className="preview-image" src={props.image} alt={props.id} />
      <Link to={`/${props.id}`}>{props.title}</Link>
   </div>;
   )
}

```